src/app/components/score-chart/score-chart.component.ts

```diff
@@ -1,19 +1,64 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, Input, OnChanges, OnInit, SimpleChanges } from '@angular/core';
 import * as Highcharts from 'highcharts';
+import more from 'highcharts/highcharts-more';
+import { parseISO } from 'date-fns';
+import { mean } from 'lodash';
+
+more(Highcharts);
 
 @Component({
   selector: 'app-score-chart',
   templateUrl: './score-chart.component.html',
   styleUrls: ['./score-chart.component.sass']
 })
-export class ScoreChartComponent implements OnInit {
+export class ScoreChartComponent implements OnChanges {
 
-  private options: any = {};
+  @Input()
+  public historicalData: string[];
 
-  constructor() { }
+  @Input()
+  public label: string;
 
-  ngOnInit(): void {
-    Highcharts.chart('container', this.options);
-  }
+  @Input()
+  public currentValue = '- -';
```

Comment:
I don't think we need default parameters (if TSLint is complaining, disable the flag?)
Moreover, the default value is of the wrong type (should be number?)

Reply:
The TSLint is not really complaining, I just wanna show -- as the default value when there is no score for day T yet. But I guess I could also move the logic to the html file.

Reply:
I agree to move the "rendering logic" to the HTML file; `currentValue` can be an optional number, and the HTML file can render the "default value" (`--`) if no data is available today:

```typescript
@input
public currentValue?: number
```